subproject-lifecycle.md

```diff
@@ -0,0 +1,121 @@
+
+1. Establish a relation with a SIG that takes patronage over your subproject idea. This SIG will become subproject's owner.
+   - Either get a verbal agreement on a SIG meeting, documented in SIG meeting notes
+   - Or file an issue in open-services-group/community with subproject details and get a SIG lead approval on the issue
+2. Communicate the transfer to all the current owners of the workstreams
+3. There must be existing GitHub repository for the workstream
+4. Declare project owners in [OWNERS file](adr/0000-use-OWNERS-rather-than-github-teams.md)
+
+## Process
+
+### Add a new subproject
+
+1. Fork `open-services-group/community`
+2. Checkout a feature branch named after your project
+3. Update [`sigs.yaml`](./sigs.yaml) file with following:
+    1. Locate the SIG entry in `sigs` top level key, it has a property `dir: sig-<NAME>`.
+    2. If your subproject has regular meetings, update the `.meetings` section with:
+
+       ```yaml
+       meetings:
+         - description: "Description containing the subproject name"
+           day: "[Monday,Tuesday,...]"
+           time: "HH:MM"
+           tz: "[UTC|ET (East Time)|CET (Central Europe Time)|...]"
+           frequency: "[weekly|bi-weekly]"
+           url: https://meet.google.com/...
+       ```
+
+    3. Add your subproject to the `.subprojects` section:
+
+       ```yaml
+       subprojects:
+         - name: <NAME>
+           owners:
+             - https://raw.githubusercontent.com/<ORG>/<REPO>/main/OWNERS  # Do not panic if the repository doesn't exist yet
+       ```
+
+4. Update [`github-config.yaml`](./github-config.yaml) file:
+    1. If you are creating a brand new subproject, please initialize a new repository as:
+
+       ```yaml
+       orgs:
+         open-services-group:
+           repos:
+             <REPOSITORY_NAME>:
+               description: Describe the subproject in one sentence.
+               has_projects: false
+               has_wiki: false
+               ...
+       ```
+
+    2. Newly created repositories needs to grant `admin` role to `SrcOps` team. Add a new line to `.teams.SourceOps.repos`:
+
+       ```yaml
+       orgs:
+         open-services-group:
+           teams:
+             SourceOps:
+               ...
+               repos:
+                 <REPOSITORY_NAME>: admin
+       ```
+
+    3. Optionally you can create new GitHub team to maintain this repository:
+
+       ```yaml
+       orgs:
+         open-services-group:
+           teams:
+             sp-<NAME>-maintainers:
+               description: Maintainers Team for the <NAME> sub-project
+               maintainers:
+                 - list of GitHub handles
+               members:
+               privacy: closed
+               repos:
+                 <REPOSITORY_NAME>: admin
+       ```
+
+5. Run the generators via `podman run --rm --volume $(pwd):/workdir:Z quay.io/open-services-group/community-tooling:v0.1.0-dev` command
```

Comment:
We could re-use the [Makefile](https://github.com/thoth-station/core/blob/master/community/Makefile) in thoth-station/core/community to help with the generator and label-sync commands... although that's probably for a different PR

Reply:
`podman` command line is good enoug for me :)